src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs

```diff
@@ -14,8 +15,9 @@ namespace Microsoft.ML.Transforms.Text
     /// Produces a bag of counts of ngrams (sequences of consecutive words) in a given text.
     /// It does so by building a dictionary of ngrams and using the id in the dictionary as the index in the bag.
     /// </summary>
-    public sealed class WordBagEstimator : TrainedWrapperEstimatorBase
+    public sealed class WordBagEstimator : IEstimator<ITransformer>
```

Comment:
The `TrainedWrapperEstimatorBase` was meant to facilitate the translation from the legacy `IDataTransform` interface to the new `ITransformer` interface. It is intended to be a temporary situation; however, by making this class descend from that abstract base class, that temporary shim would have had to become permanent -- you cannot "undescend" from a class without it being a breaking change in your API. Therefore we would have had to live with it forever.

So we had to duplicate the logic that existed in that base class. Fortunately there were only two subclasses remaining (both in this file, as it happened), so it was not such a big problem.

Reply:
This is a really good idea! until we do the proper translation

---
In reply to: [262299912](https://github.com/dotnet/machinelearning/pull/2842#discussion_r262299912) [](ancestors = 262299912)